pkg/registries/registries.go

```diff
@@ -139,7 +166,12 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegi
 		reg.Insecure = true
 		for i := range config.Registries {
 			reg := &config.Registries[i]
-			if scopeMatchesRegistry(reg.Location, insecureReg) {
+			scope := reg.Location
+			// If Prefix is a valid wildcard entry, we set scope to Prefix
+			if isValidWildcard(reg.Prefix) {
+				scope = reg.Prefix
+			}
+			if scopeMatchesRegistry(scope, insecureReg) {
```

Comment:
This no longer works. We had `scopeMatchesRegistry` because `insecureReg` and `blockedReg` were only allowed to be registry host[:port]. Now that insecure/blocked can be host:port or a wildcard, we need a `scopeIsSubscopeOf` (for a case of `insecure: *.example.com` + ICSP Source: `foo.example.com/namespace`)— handling both non-wildcard hosts vs. non-wildcard namespaces, wildcard hosts vs. non-wildcard namespaces, wildcard hosts vs. more specific wildcard hosts.

`insecureReg` should now be renamed to something like `insecureScope`; and so should probably the function parameter.

Also for blocked.

Reply:
Added `isSubScopeOfReg`